scripts/link-checker.js

```diff
@@ -227,7 +227,16 @@ async function checkLinks() {
 
         let matchesHash = true;
         if (hash) {
-          matchesHash = !!matchingPage?.ids.includes(hash);
+          // On some websites, the ids may not exactly match the hash included
+          // in the link.
+          // For e.g. GitHub will prepend client facing ids with their own
+          // calculated value. A heading in a README for example could be
+          // Foo bar, navigated to with https://github.com/foo/bar#foo-bar,
+          // but GitHub calculates the actual markup id included in the document
+          // as being user-content-foo-bar for its own page processing purposes.
+          //
+          // See https://github.com/w3c/aria-practices/issues/2809
+          matchesHash = !!matchingPage?.ids.some((id) => id.includes(hash));
```

Comment:
I think this condition is not sufficiently restrictive. It makes it too easy for me (and others) to create href attributes that are not valid. I really want to be confident that our content does not contain invalid fragment ids. Even though there is the possibility of more maintenance to this code, I'd rather have the condition be specific to GitHub pages and the "user-content-" prefix. It does not seem likely we will encounter this problem on other pages referenced by the APG.

Over time, if it turns out GitHub is wishywashy in there approach or if this turns out to be a problem with other sites, we can decide how to address those other scenarios when they arise.

Reply:
@mcking65 I've updated the code to account specially account for this GitHub scenario.

I also set it up to handle any others in the future using the `.link-checker.js` file which seems to handle some config settings.

Reply:
cc @alflennik I also removed `/^https:\/\/github\.com\/.*\/wiki\//` from `ignoreHashesOnExternalPagesMatchingRegex` as the same situation would happen on those wiki pages.